### PR TITLE
Fix VMware OVF properties copy from template

### PR DESCRIPTION
**Pull request by nvazquez**
### Description

For backward compatibility, prior to Vmware 6.5 use EDIT operation instead of ADD when copying template vApp information 

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan help

**Comment by nvazquez** [comment]
@blueorangutan test centos7 vmware-60u2

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by borisstoyanov** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test centos7 vmware-60u2

**Comment by rhtyd** [comment]
@nvazquez can you see the vapp failure? cc @borisstoyanov

File: plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java [hunk 1]
```diff
@@ -2707,13 +2710,14 @@ private String getPropertyValue(OVFPropertyTO prop) {
...
      */
-    protected List<VAppProductSpec> copyVAppConfigProductSectionFromOVF(VmConfigInfo vAppConfig) {
+    protected List<VAppProductSpec> copyVAppConfigProductSectionFromOVF(VmConfigInfo vAppConfig, boolean useEdit) {
         List<VAppProductInfo> productFromOvf = vAppConfig.getProduct();
         List<VAppProductSpec> specs = new ArrayList<>();
         for (VAppProductInfo info : productFromOvf) {
             VAppProductSpec spec = new VAppProductSpec();
             spec.setInfo(info);
-            spec.setOperation(ArrayUpdateOperation.ADD);
+            s_logger.info("Procuct info KEY " + info.getKey());
```
Comment (sureshanaparti) [inline plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java:2719]: Product - typo

File: plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java [hunk 2]
```diff
@@ -2725,16 +2729,19 @@ private String getPropertyValue(OVFPropertyTO prop) {
...
         vmConfigSpec.setInstallBootStopDelay(vAppConfig.getInstallBootStopDelay());
         vmConfigSpec.setInstallBootRequired(vAppConfig.isInstallBootRequired());
         vmConfigSpec.setIpAssignment(vAppConfig.getIpAssignment());
         vmConfigSpec.getOvfEnvironmentTransport().addAll(vAppConfig.getOvfEnvironmentTransport());
-        vmConfigSpec.getProduct().addAll(copyVAppConfigProductSectionFromOVF(vAppConfig));
-        vmConfigSpec.getProperty().addAll(copyVAppConfigPropertySectionFromOVF(vAppConfig, ovfProperties));
-        vmConfigSpec.getOvfSection().addAll(copyVAppConfigOvfSectionFromOVF(vAppConfig));
+
+        // For backward compatibility, prior to Vmware 6.5 use EDIT operation instead of ADD
+        boolean useEditOperation = hyperHost.getContext().getServiceContent().getAbout().getApiVersion().compareTo("6.5") < 1;
```
Comment (sureshanaparti) [inline plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java:2741]:
```suggestion
        boolean useEditOperation = ((HostMO) hyperHost).getHostAboutInfo().getApiVersion().compareTo("6.5") < 1;
```

**Review by rhtyd — APPROVED** [review]

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by borisstoyanov** [comment]
@blueorangutan test centos7 vmware-60u2

**Comment by borisstoyanov** [comment]
ContextSuite context=TestVAppsVM>:setup | Error | 44.52 | test_vm_life_cycle.py
-- | -- | -- | --


This seems like marvin related failure, it's basically failing to setup the test. @nvazquez can you have a look pls

**Comment by rhtyd** [comment]
@blueorangutan test centos7 vmware-60u2

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by borisstoyanov** [comment]
@blueorangutan test centos7 vmware-60u2

**Comment by rhtyd** [comment]
@blueorangutan test centos7 vmware-67u3

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan package

**Review by borisstoyanov — APPROVED** [review]
LGTM, tested OVF properties and advanced vsphere capabilities features with setup of ESXi6.0, 6.5 and 6.7 hosts. Other than the known issues there were no issues with this bugfix.

**Comment by nvazquez** [comment]
@rhtyd can you give this PR a final review? Test fixes added after your review

**Comment by rhtyd** [comment]
Lgtm!

**Pull request merged**